### PR TITLE
Less pip install issue51

### PR DESCRIPTION
**Pull request by zhouxinyu0723**

### CODE REVIEW

**Comment by PhilippvK** [comment]
Please check the linter issues. Most of the stuff can be resolved using `black -l 120 mlonmcu`

File: .gitignore [hunk 1]
```diff
@@ -111,3 +111,6 @@ ENV/
 out/
 deps/
 environment.yml
+
+# requirements addtion
+requirements_addition.txt
```
Comment (PhilippvK) [inline .gitignore:116]: Can be removed if file is written to env dir

File: setup.py [hunk 2]
```diff
@@ -38,7 +39,10 @@ def resource_files(directory):
 with open("HISTORY.md") as history_file:
     history = history_file.read()
 
-requirements = []
+requirements = gen_requirements.join_requirements()
+extra_require = {
+    piece: deps for piece, (_, deps) in requirements.items() if piece not in ("all", "core")
```
Comment (PhilippvK) [inline setup.py:44]: This is great!

**Review by PhilippvK — APPROVED** [review]

**Comment by PhilippvK** [comment]
`mlonmcu/read_write_file_lock/.nfs80fc025d08848a620000031f ` should probably be removed

**Comment by zhouxinyu0723** [comment]
should I fix the black and .gitignore and make a new commit and a new PR?

**Comment by PhilippvK** [comment]
No, just add new commits to the pr branch

**Comment by PhilippvK** [comment]
@zhouxinyu0723 Thank you. Black issues are resolved now, while anything which is not Flake8 (a bit stricter linter) compliant can be found here and needs to be fixed manually:

https://github.com/tum-ei-eda/mlonmcu/pull/80/checks?check_run_id=9005475013

**Comment by PhilippvK** [comment]
Look very good @zhouxinyu0723. Thank you!

**Pull request merged**